### PR TITLE
Cu 86bx0njw8/be paginate upcoming activities endpoint return 5 items per page starting from most recent

### PR DESCRIPTION
**Pull request by Don-christo**
**Description of PR**

Paginated upcoming activities endpoint.

How to test:
- Add dummy data to your local db using the `createGroup` function. With the data you've created, you can use them to test the pagination function.

---

Please check all that apply.

#### Checklist:

- [x] Rebased with master
- [x] PR Description provided
- [x] Did you communicate these changes with the affected teams?
- [ ] Did unit tests/integration tests/E2E tests get properly added/updated?
- [ ] Is this change production-ready including testing, feature toggles, configurations, and deployment considerations?

#### Frontend Checklist:

- [ ] Is this responsive and mobile friendly?
- [ ] Did you follow the [ARIA specs](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA)

#### Backend Checklist:

- [x] Does the endpoint return the correct success/error format?

---

**Screenshots**  
_add relevant screenshots here_

---

**Test Coverage**  
_add relevant screenshots here_

### CODE REVIEW

**Comment by KristofaJosh** [comment]
Task linked: [CU-86bx0njw8 BE-paginate upcoming Activities endpoint. Return 5 items per page starting from most recent](https://app.clickup.com/t/86bx0njw8)

**Review by pelzman — DISMISSED** [review]
Weldone IK

File: apps/server/src/controllers/userControllers/getUpcomingActivities.ts
Comment (KristofaJosh) [inline apps/server/src/controllers/userControllers/getUpcomingActivities.ts:1]: I think you should prefer this approach with sequelize:

we can discuss this

```
...
const offset = (page - 1) * pageSize;
const limit = pageSize;

try {
     const results = await ModelName.findAndCountAll({ offset,   limit, order: [['createdAt', 'DESC']]});
     return {
            totalItems: results.count,
            totalPages: Math.ceil(results.count / pageSize),
            currentPage: page,
            pageSize: limit,
            data: results.rows,
        };
```

File: apps/server/src/controllers/userControllers/userUpdateKyc.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+      proof_of_address_doc,
+    } = validationResult.data
+
+    const files = req.files as {
+      [fieldname: string]: Express.Multer.File[]
+    }
+    console.log(files)
+
+    let _identification_doc = identification_doc ?? user.identification_doc
+    let _proof_of_address_doc =
```
Comment (KristofaJosh) [inline apps/server/src/controllers/userControllers/userUpdateKyc.ts:42]: This shouldn't be here

**Review by KristofaJosh — CHANGES_REQUESTED** [review]

File: apps/server/src/controllers/userControllers/userUpdateKyc.ts (hunk 1 above)
Comment (Don-christo) [reply apps/server/src/controllers/userControllers/userUpdateKyc.ts:42]: You mean the file??

File: apps/server/src/controllers/userControllers/getUserProfileDetails.ts
Comment (KristofaJosh) [inline apps/server/src/controllers/userControllers/getUserProfileDetails.ts:1]: not related to this ticket

File: apps/server/src/controllers/userControllers/getUpcomingActivities.ts [hunk 2]
```diff
@@ -87,10 +87,50 @@ export const getUpcomingUserActivities = async (
...
+        currentPage: page,
+        pageSize: limit,
+        data: results.rows,
+      }
+      const queryStringData = Object.entries(responseData)
+        .map(
+          ([key, value]) =>
+            `${key}=${encodeURIComponent(JSON.stringify(value))}`
+        )
+        .join('&')
```
Comment (KristofaJosh) [inline apps/server/src/controllers/userControllers/getUpcomingActivities.ts:120]: Use this:

https://www.npmjs.com/package/query-string

File: apps/server/src/controllers/userControllers/getUpcomingActivities.ts
Comment (KristofaJosh) [inline apps/server/src/controllers/userControllers/getUpcomingActivities.ts:1]: re check this and make the necessary fixes, given the new package I have shared with you


Good job so far

**Review by KristofaJosh — CHANGES_REQUESTED** [review]

**Review by KristofaJosh — APPROVED** [review]

**Review by KristofaJosh — DISMISSED** [review]

**Review by KristofaJosh — APPROVED** [review]

**Pull request merged**